src/Smart.Design.Razor/TagHelpers/Header/HeaderTagHelper.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.TagHelpers;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+using Smart.Design.Razor.TagHelpers.Constants;
+
+namespace Smart.Design.Razor.TagHelpers.Header;
+
+[HtmlTargetElement(TagNames.HeaderTagName)]
+public class HeaderTagHelper : TagHelper
+{
+    private readonly IHeaderHtmlGenerator _headerHtmlGenerator;
+
+    /// <summary>
+    /// Home page url from appsettings context
+    /// </summary>
+    public string HomePageUrl { get; set; }
+
+    /// <summary>
+    /// list of available languages
+    /// </summary>
+    public Dictionary<string, string> LanguagesAndLinks { get; set; } = new Dictionary<string, string>();
+
+    /// <summary>
+    /// Current interface language
+    /// </summary>
+    public string CurrentLanguage { get; set; }
+
+    /// <summary>
+    /// Current user full name
+    /// </summary>
+    public string UserName { get; set; }
+
+    /// <summary>
+    /// Path to the current user avatar
+    /// </summary>
+    public string AvatarPath { get; set; }
+
+    /// <summary>
+    /// Dictionnary of label and links items for the drop down menu
```

Comment:
```suggestion
    /// Dictionary of label and links items for the drop down menu
``` 
(just to be a pain haha)